### PR TITLE
fix(LEMS-2738): Convert Expression Editor Answer keys to strings

### PR DESCRIPTION
**Pull request by anakaren-rojas**
## Summary:
- Converts expression editor keys to strings
- Updates `onDelete` handler to use index instead of key

Issue: LEMS-2738

## Test plan:

https://github.com/user-attachments/assets/60424f5a-f654-4afc-901f-de47505e053c

### CODE REVIEW

File: packages/perseus-editor/src/widgets/expression-editor.tsx [hunk 1]
```diff
@@ -565,6 +570,7 @@ class AnswerOption extends React.Component<
 
     handleImSure = () => {
         this.props.onDelete();
+        this.handleCancelDelete();
```
Comment (catandthemachines) [inline packages/perseus-editor/src/widgets/expression-editor.tsx:573]: Ooo, what's this do and why is it here?

**Review by catandthemachines — APPROVED** [review]
I have one comment on a line that looked odd that I have questions about. But once that's resolved it looks good to go.

File: packages/perseus-editor/src/widgets/expression-editor.tsx (hunk 1 above)
Comment (anakaren-rojas) [reply packages/perseus-editor/src/widgets/expression-editor.tsx:573]: Its an existing function that closes the "I'm sure" button, I added it so it wouldn't stay open once someone deletes the current item

Comment (anakaren-rojas) [reply packages/perseus-editor/src/widgets/expression-editor.tsx:573]: Here is the before: 
https://github.com/user-attachments/assets/8838f716-dfd4-4aa7-995d-b15f4c5aa22a

Here is the after: 

https://github.com/user-attachments/assets/8766dcda-036d-4298-906f-018aec5b0859

**Pull request merged**